pom.xml

```diff
@@ -126,10 +126,10 @@
     <commons-configuration.version>1.10</commons-configuration.version>
     <commons-exec.version>1.1</commons-exec.version>
     <commons-io.version>2.12.0</commons-io.version>
-    <commons-pool2.version>2.11.1</commons-pool2.version>
+    <commons-pool2.version>2.12.0</commons-pool2.version>
```

Comment:
this is a transitive dependency of commons-dbcp2. do we really need to explicitly declare it?

Reply:
commons-pool2 was made to be declared explicitly as part of : [HIVE-26423](https://issues.apache.org/jira/browse/HIVE-26423) & [PR-3472](https://github.com/apache/hive/pull/3472)
where the reasoning was that hive uses commons-pool2 directly and to prevent any kind of runtime/packaging issues, we need to declare it explicitly.

Reply:
if you are trying to `make commons-pool2 version consistent` why do you explicitly define the dependency version in ql module and instead of dependency-management in root pom?

Reply:
commons-pool2 was defined in ql module in the same PR where it was declared explicitly - [PR-3472](https://github.com/apache/hive/pull/3472) . But, i believe that would be because mentioning in dependency-management in root pom would only make sure that version of commons-pool2 coming in transitively is kept consistent and not explicitly make commons-pool2 as direct dependency of hive which was required then. 
But even i am not sure that why was it declared only in ql and not other modules such as standalone-metastore where commons-dbcp2 is being pulled in.

Reply:
What I meant was to declare the version in dependency management, so you won't get different dependency versions in multiple Hive modules, same as it's done for `commons-dbcp2`

Reply:
`to prevent any potential runtime issues` - how could it cause runtime issues, scope of transitive dependency is compile, not provided. I don't get what [HIVE-26423](https://issues.apache.org/jira/browse/HIVE-26423) is supposed to fix.

Reply:
Yeah, it is safe to remove as far as I can tell. It was a downstream issue in which Compaction was failing at runtime because "commons-pool2" was not getting packaged. I have no idea why I decided to put this upstream rather than just keep it downstream.

Reply:
Thank you @jfsii for the information. I'll make the changes to remove commons-pool2 as an explicit dependency & upgrade only commons-dbcp2 in the same PR and also update the JIRA to reflect the same.